### PR TITLE
test: test for default meta feature

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
Adding tests for all widgets supporting this feature

### CODE REVIEW

File: app/client/cypress/support/widgetCommands.js [hunk 1]
```diff
@@ -1495,3 +1495,195 @@ Cypress.Commands.add("selectTableAndReset", () => {
     cy.wrap(item).should("contain.text", "#1");
   });
 });
+
+Cypress.Commands.add("selectSwitchGroupAndReset", () => {
```
Comment (Rishabh-Rathod) [inline app/client/cypress/support/widgetCommands.js:1499]: Do we need to make `selectSwitchGroupAndReset` as cypress command? 
Can we not keep it a javascript function and call it `action` function.

Comment (NandanAnantharamu) [reply app/client/cypress/support/widgetCommands.js:1499]: we can do that as function , i will replace those as functions only

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+      );
+      // Bind to stored value above
+      cy.openPropertyPane(WIDGET.TEXT);
+      cy.updateCodeInput(PROPERTY_SELECTOR.text, testConfig.textBindingValue);
+    });
+
+    it("3. Publish the app and check the reset action", () => {
+      // Set onClick action, storing value
+      cy.PublishtheApp();
+      testConfig.action();
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: done

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+  cy.wait(1000);
+  cy.get(".t--text-widget-container").each((item, index, list) => {
+    cy.wrap(item).should("not.contain.text", "BLUE");
+  });
+}
+
+function selectTabAndReset() {
+  cy.get(".t--tabid-tab2").click({ force: true });
+  cy.wait(1000);
+  cy.get(".t--text-widget-container").each((item, index, list) => {
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: done

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+  cy.wait(1000);
+}
+
+function audioRecorderWidgetAndReset() {
+  cy.get(".t--text-widget-container").each((item, index, list) => {
+    cy.wrap(item).should("contain.text", "true");
+  });
+  cy.get("button:contains('Submit')").click({ force: true });
+  cy.wait(1000);
+}
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: This i am only validating the msg part, i need to check with Saptami once.If its a bug i will raise and comment these 2 widgets

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rishabh-Rathod) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: @NandanAnantharamu seems like `action` is replaced with `assertWidgetReset` is places where it was not supposed to also. Can you please check once again.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: https://github.com/appsmithorg/appsmith/issues/14693
https://github.com/appsmithorg/appsmith/issues/14692
These bugs are open hence commenting Audio/AudioRecorder widget tests.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+
+function selectSwitchAndReset() {
+  cy.get(".bp3-control-indicator")
+    .last()
+    .click({ force: true });
+  cy.get(".t--switch-widget-active").should("not.exist");
+  cy.get("button:contains('Submit')").click({ force: true });
+  cy.wait(1000);
+  cy.get(".t--toast-assertWidgetReset span").contains("success");
```
Comment (Rishabh-Rathod) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js:218]: `t--toast-assertWidgetReset` is this correct selector?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rishabh-Rathod) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: Could you also add same information as comment that we need to add test case once above issues are resolved.

[the remark this answers is not in the document]
Comment (Rishabh-Rathod) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/AllWidgets_default_meta_spec.js]: I see all test are passing, we need not to block this PR due to the comment we could add it later in separate PR.

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Pull request merged**